### PR TITLE
transform feedback separated

### PR DESCRIPTION
**Pull request by trungtle**
@shrekshao @pjcozzi Ready for review. I didn't use a query to get the TRANSFORM_FEEDBACK_PRIMITIVES in the 2nd draw pass since WebGL2 can only get the query's result asynchronously. I opted to just hardcoded the VERTEX_COUNT length instead.

### CODE REVIEW

File: samples/transform_feedback_separated.html [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        }
+    </script>
+
+    <!-- This fragment shader will not be called since it is used when the rasterizer is disabled. -->
+    <script id="fs-transform" type="x-shader/x-fragment">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        in vec4 color;
```
Comment (pjcozzi) [inline samples/transform_feedback_separated.html]: I would call this `v_color` and then call the output color just `color`.

File: samples/transform_feedback_separated.html [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            vColor = color;
+        }
+    </script>
+
+    <script id="fs-feedback" type="x-shader/x-fragment">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        in vec4 vColor;
```
Comment (pjcozzi) [inline samples/transform_feedback_separated.html]: Same comment.

File: samples/transform_feedback_separated.html [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            document.getElementById('info').innerHTML = 'WebGL 2 is not available.  See <a href="https://www.khronos.org/webgl/wiki/Getting_a_WebGL_Implementation">How to get a WebGL 2 implementation</a>';
+            return;
+        }
+
+        // -- Init Program
+        var programTransform = createProgram(gl, getShaderSource('vs-transform'), getShaderSource('fs-transform'));
+
+        var varyings = ['gl_Position', 'color'];
+        gl.transformFeedbackVaryings(programTransform, varyings, gl.SEPARATE_ATTRIBS);
+        gl.linkProgram(programTransform); // Relink program for transform feedback varyings
```
Comment (pjcozzi) [inline samples/transform_feedback_separated.html]: Instead of relinking here, we should include all the required code from `createProgram` inline here.  Otherwise, this is too hard to follow.  See `transform_feedback_interleaved`.

File: samples/transform_feedback_separated.html [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        var mvpLocation = gl.getUniformLocation(programTransform, 'MVP');
+
+        var PROGRAM_TRANSFORM = 0;
+        var PROGRAM_FEEDBACK = 1;
+        var programs = [programTransform, programFeedback];
+
+        // -- Init Buffer
+
+        var VERTEX_COUNT = 6;
+        var vertices = new Float32Array([
```
Comment (pjcozzi) [inline samples/transform_feedback_separated.html]: Are these positions?  If so name them as such since it is more precise.

File: samples/transform_feedback_separated.html [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        gl.bindBuffer(gl.ARRAY_BUFFER, buffers[BufferType.COLOR]);
+        gl.vertexAttribPointer(vertexColorLocationFeedback, 4, gl.FLOAT, false, 0, 0);
+        gl.enableVertexAttribArray(vertexColorLocationFeedback);
+
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+        gl.bindVertexArray(null);
+
+        // -- Render
+
+        render();
```
Comment (pjcozzi) [inline samples/transform_feedback_separated.html]: Remove the `render` function (it is not needed and disrupts the flow) and include the code in this function.

**Comment by pjcozzi** [comment]
@shrekshao do you have any feedback here?

**Comment by trungtle** [comment]
@pjcozzi I push dd9aa10 with updates according to your comments

File: samples/transform_feedback_separated.html [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+        // -- Init Canvas
+        var canvas = document.createElement('canvas');
+        canvas.width = Math.min(window.innerWidth, window.innerHeight);
+        canvas.height = canvas.width;
+        document.body.appendChild(canvas);
+
+        // -- Init WebGL Context
+        var gl = canvas.getContext('webgl2', { antialias: false });
+        if (!gl) {
+            gl = canvas.getContext('experimental-webgl2', { antialias: false });
```
Comment (kenrussell) [inline samples/transform_feedback_separated.html]: Note: there's been agreement among the browser vendors that we're not going to expose an 'experimental-webgl2' context type. They'll all expose 'webgl2' behind a flag until it's ready to ship, at which point it'll be exposed unconditionally. So throughout these samples, any fetching of the 'experimental-webgl2' context type can and should be removed (since we shouldn't encourage web developers to follow this pattern).

Comment (trungtle) [inline samples/transform_feedback_separated.html]: Opened issue #37 for this

Comment (pjcozzi) [inline samples/transform_feedback_separated.html]: Sounds great!  Thanks for the input, @kenrussell!

**Comment by pjcozzi** [comment]
Looks good.  Made a few tweaks in 8ccf5cce65f958d9fdf067961a3c9f50883128bd.
